test.py

```diff
@@ -0,0 +1,191 @@
+    # 16, 17, 18, 19: 0b0100xx
+    charsD = ''.join(sorted('QRST'))
+    # 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62: 0bxxxx10
+    charsE = ''.join(sorted('CGKOSWaeimquy26-'))
+    # 0, 16, 32, 48: 0bxx0000
+    charsF = ''.join(sorted('AQgw'))
+    expected = [charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsD, charsAll, charsE, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsF]
+    spreadTest(slugid.v4, expected)
+
+
+def spreadTest(generator, expected):
+    """ `spreadTest` runs a test against the `generator` function, to check that
+    when calling it 64*40 times, the range of characters per string position it
+    returns matches the array `expected`, where each entry in `expected` is a
+    string of all possible characters that should appear in that position in the
+    string, at least once in the sample of 64*40 responses from the `generator`
+    function """
+    # k is an array which stores which characters were found at which
+    # positions. It has one entry per slugid character, therefore 22 entries.
+    # Each entry is a dict with a key for each character found, and its value
+    # as the number of times that character appeared at that position in the
+    # slugid in the large sample of slugids generated in this test.
+    k = [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
+
+    # Generate a large sample of slugids, and record what characters appeared
+    # where...  A monte-carlo test has demonstrated that with 64 * 20
+    # iterations, no failure occurred in 1000 simulations, so 64 * 40 should be
+    # suitably large to rule out false positives.
+    for i in range(0, 64 * 40):
+        slug = generator()
+        assert len(slug) == 22
+        for j in range(0, 22):
+            if slug[j] in k[j]:
+                k[j][slug[j]] = k[j][slug[j]] + 1
+            else:
+                k[j][slug[j]] = 1
+
+    # Compose results into an array `actual`, for comparison with `expected`
+    actual = []
+    for j in range(0, len(k)):
+        actual.append('')
+        for a in k[j].keys():
+            if k[j][a] > 0:
+                actual[j] += a
+        # sort for easy comparison
+        actual[j] = ''.join(sorted(actual[j]))
+
+    assert arraysEqual(expected, actual), "In a large sample of generated slugids, the range of characters found per character position in the sample did not match expected results.\n\nExpected: " + str(expected) + "\n\nActual: " + str(actual)
+
+def arraysEqual(a, b):
+    """ `arraysEqual` checks arrays `a` and `b` for equality"""
+    if a == b:
+        return True
+    if a is None or b is None:
+        return False
+    if len(a) != len(b):
+        return False
+    for i in range (0, len(a)):
+        if a[i] != b[i]:
+            return False
+    return True
```

Comment:
This is a nice shorter version, thanks!
I spotted `cmp` built in function got removed in python 3, but if we choose to support python 3 later, there is the [cmp_to_key](https://docs.python.org/3/library/functools.html#functools.cmp_to_key) that should help us migrate.